hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerV2Impl.java

```diff
@@ -431,15 +464,21 @@ public void scrubPipeline(ReplicationType type, ReplicationFactor factor)
    * Schedules a fixed interval job to create pipelines.
    */
   @Override
-  public void startPipelineCreator() {
+  public void startPipelineCreator() throws NotLeaderException {
+    if (!scmhaManager.isLeader()) {
```

Comment:
when scm leader changes, how to ensure the creator and scrubber threads are started on the new leader and stopped on the old leader?

Reply:
My thought:
1. Every SCM starts a pipeline creator thread. So I remove checkLeader in startPipelineCreator
2. For triggerPipelineCreator, if we decide to let follower go thru safemode once it steps up to be leader, we can have checkLeader there as well.